src/main/java/Duke.java

```diff
@@ -51,29 +51,52 @@ public static void main(String[] args) {
                 }
             } else {
                 if (input.contains ("todo")) {
-                    Todo todo = new Todo (input.substring(5));
-                    lst.add (todo);
-                    pendingTask ++;
-                    System.out.println ("Got it. I've added the following task:\n" +
-                            todo + "\nYou now have " + pendingTask + " task in the list");
+                    //todo request format: todo<space><task>
+                    try {
+                        String task1 = input.substring(5);
+                        if (task1.isEmpty()) {
+                            System.out.println ("OOOPS!! Cannot have empty todo request!!!");
+                        } else {
+                            Todo todo = new Todo(task1);
+                            lst.add(todo);
+                            pendingTask++;
+                            System.out.println("Got it. I've added the following task:\n" +
+                                    todo + "\nYou now have " + pendingTask + " task in the list");
+                        }
+                    } catch (Exception e) {
+                        System.out.println ("Huh? I do not understand this todo request:/");
+                    }
                 } else if (input.contains ("deadline")) {
-                    int taskIndex = input.indexOf("/");
-                    int byIndex   = taskIndex + 4;
-                    Deadline deadline = new Deadline (input.substring (9, taskIndex),
-                            input.substring (byIndex));
-                    lst.add (deadline);
-                    pendingTask++;
-                    System.out.println ("Got it. I've added the following task:\n" +
-                            deadline + "\nYou now have " + pendingTask + " task in the list");
+                    //deadline request format: deadline<space><task><space></><by when>"
+                    try {
+                        int taskIndex = input.indexOf("/");
+                        int byIndex = taskIndex + 4;
+                        Deadline deadline = new Deadline(input.substring(9, taskIndex),
+                                input.substring(byIndex));
+                        lst.add(deadline);
+                        pendingTask++;
+                        System.out.println("Got it. I've added the following task:\n" +
+                                deadline + "\nYou now have " + pendingTask + " task in the list");
+                    } catch (Exception e) {
+                        System.out.println ("Huh? This deadline request does not make sense");
+                    }
                 } else if (input.contains ("event")) {
-                    int taskIndex = input.indexOf("/");
-                    int atIndex   = taskIndex + 4;
```

Comment:
Perhaps extract the values to a meaningful keyword (e.g. task description, date)?